onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -1790,6 +1790,23 @@ class PlannerImpl {
     // 2. determing following things:
     //    a. which node need to generate notification
     //    b. which node need to trigger downstream
+#ifdef ENABLE_TRAINING
+    InlinedHashMap<NodeIndex, size_t> node_index_2_toposort_index;
+    const std::vector<NodeIndex>& topo_sort = graph_viewer_.GetNodesInTopologicalOrder(context_->GetExecutionOrder());
+    node_index_2_toposort_index.reserve(topo_sort.size());
+    size_t yieldOp_index_in_toposort = topo_sort.size();
+    for (size_t i = 0; i < topo_sort.size(); i++) {
+      const Node* node = graph_viewer_.GetNode(topo_sort[i]);
+      node_index_2_toposort_index[node->Index()] = i;
+      if (node->OpType() == "YieldOp") yieldOp_index_in_toposort = i;
```

Comment:
assert that only 1 yieldop node?

Reply:
I tried, but this code path is also executed by the models without any yieldOp